scripts/reportInstalls.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { promisify } from 'util';
+import { exec } from 'child_process';
+import * as appInsights from 'applicationinsights';
+import { DEFAULT_AIKEY } from '../packages/salesforcedx-utils-vscode/src/constants';
+
+const promisifiedExec = promisify(exec);
+
+interface InstallTelemetryData {
+  // To align with the properties property for appInsightsClient.trackEvent
+  [key: string]: string;
+  install: string;
+  ratingcount: string;
+  marketplace: string;
+  'common.extname': string;
+}
+
+// These properties match the values provided in the statistics array from the
+// vsce cli output
+interface InstallTelemetryDataMS extends InstallTelemetryData {
+  averagerating: string;
+  trendingdaily: string;
+  trendingmonthly: string;
+  trendingweekly: string;
+  updateCount: string;
+  weightedRating: string;
+  downloadCount: string;
+}
+
+type TelemetryData = InstallTelemetryData | InstallTelemetryDataMS;
+
+// Set to true for local development
+const DRY_RUN = false;
+const extensionsToTrack = [
+  'salesforcedx-vscode-core',
+  'salesforcedx-vscode-apex',
+  'salesforcedx-vscode-apex-debugger',
+  'salesforcedx-vscode-apex-replay-debugger',
+  'salesforcedx-vscode-lightning',
+  'salesforcedx-vscode-lwc',
+  'salesforcedx-vscode-visualforce',
+  'salesforcedx-vscode',
+  'salesforcedx-vscode-expanded',
```

Comment:
I think this one might be causing errors - I'm not seeing anything past this in telemetry

Reply:
Hopefully fixed by adding a flush to the app insight client before exit.